### PR TITLE
AlignComponent algorithm

### LINKED ISSUE #13297: Align Component algorithm
This algorithm will take the output of [GetDetOffsetsMultiPeaks](http://docs.mantidproject.org/nightly/algorithms/GetDetOffsetsMultiPeaks-v1.html) in the form of a [Diffraction calibration workspace](http://docs.mantidproject.org/nightly/concepts/DiffractionCalibrationWorkspace.html) and minimize the difference between the `difc` of the instrument and calibration workspace by moving and rotating instrument components.

### PR DESCRIPTION
**Pull request by rosswhitfield**
Resolves #13297 

This algorithm takes the output of [GetDetOffsetsMultiPeaks](http://docs.mantidproject.org/nightly/algorithms/GetDetOffsetsMultiPeaks-v1.html) _et al_ in the form of a [Diffraction calibration workspace](http://docs.mantidproject.org/nightly/concepts/DiffractionCalibrationWorkspace.html) and minimize the difference between the `difc` of the instrument and calibration workspace by moving and rotating instrument components.

**To test:**
- Have a look at the usage examples to understand the way the algorithm works and that it gives somewhat sensible results.

[Release Notes](http://www.mantidproject.org/index.php?title=ReleaseNotes_3_7_Framework_Changes&diff=26325&oldid=26322)

### CODE REVIEW

File: Framework/Algorithms/src/CalculateDIFC.cpp [hunk 1]
```diff
@@ -71,11 +71,15 @@ void CalculateDIFC::exec() {
...
-      boost::dynamic_pointer_cast<MatrixWorkspace>(SpecialWorkspace2D_sptr(
-          new SpecialWorkspace2D(inputWs->getInstrument())));
-
-  outputWs->setTitle("DIFC workspace");
+  API::MatrixWorkspace_sptr outputWs = getProperty("OutputWorkspace");
+
+  if ((!bool(inputWs == outputWs)) ||
+      (!bool(boost::dynamic_pointer_cast<SpecialWorkspace2D>(outputWs)))) {
+    outputWs =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(SpecialWorkspace2D_sptr(
```
Comment (quantumsteve) [inline Framework/Algorithms/src/CalculateDIFC.cpp]: `SpecialWorkspace2D_sptr(new SpecialWorkspace2D(inputWs->getInstrument())` should be changed to `boost::make_shared<SpecialWorkspace2D>(inputWs->getInstrument())`

**Comment by rosswhitfield** [comment]
Good to go. Failing builds unrelated.

**Comment by peterfpeterson** [comment]
I have used this to get updated positions for NOMAD. I think it is ready to start letting users play with.

:shipit:

**Comment by FedeMPouzols** [comment]
:+1:

**Pull request merged**